tests: add exceptions for lease errors due to etcd

jibi opened the pull request:
Following up on #23334, add more exceptions for errors that seem to not be related to Cilium but rather to etcd.

Fixes: #24701

tklauser commented on test/helpers/cons.go:
```diff
@@ -242,6 +242,9 @@ const (
 	retrieveResLock              = "retrieving resource lock kube-system/cilium-operator-resource-lock" // cf. https://github.com/cilium/cilium/issues/16402#issuecomment-871155492
 	failedToRelLockEmptyName     = "Failed to release lock: resource name may not be empty"             // cf. https://github.com/cilium/cilium/issues/16402#issuecomment-985819560
 	failedToUpdateLockReqTimeout = "Failed to update lock: etcdserver: request timed out"
```
I guess `failedToUpdateLockReqTimeout` can be dropped in favor of the more generic `failedToUpdateLock` added below, given we match using `strings.Contains` https://github.com/cilium/cilium/blob/89f9525dc7ce42d81fd6ef794207c0ce996f02ab/test/helpers/utils.go#L459

jibi replied:
good shout, dropped :+1:

jibi committed:
tests: add exceptions for lease errors due to etcd

Following up on #23334, add more exceptions for errors that seem to not
be related to Cilium but rather to etcd.

Fixes: #24701
Suggested-by: André Martins <andre@cilium.io>
Signed-off-by: Gilberto Bertin <jibi@cilium.io>
